infrahub_sdk/_importer.py

```diff
@@ -2,20 +2,26 @@
 
 import importlib
 import sys
+from pathlib import Path
 from typing import TYPE_CHECKING, Optional
 
 from .exceptions import ModuleImportError
 
 if TYPE_CHECKING:
-    from pathlib import Path
     from types import ModuleType
 
+module_mtime_cache: dict[str, float] = {}
+
 
 def import_module(
     module_path: Path, import_root: Optional[str] = None, relative_path: Optional[str] = None
 ) -> ModuleType:
     import_root = import_root or str(module_path.parent)
 
+    file_on_disk = module_path
+    if import_root and relative_path:
+        file_on_disk = Path(import_root) / Path(relative_path) / module_path.name
```

Comment:
Maybe something like this `Path(import_root, relative_path, module_path.name)`?

Reply:
Ah, yes good catch. Fixed!